app/controllers/samples_controller.rb

```diff
@@ -525,37 +525,40 @@ def fill_lineage_details
     # Preserve names of the negative 'non-specific' nodes.
     # Used for the tree view and sets appropriate lineage info at each node.
 
+    missing_vals = {
```

Comment:
I think fill_lineage_details function should be in the TaxonLineage model. we should make samples_controller as simple as possible.

Reply:
good idea